controller/service.go

```diff
@@ -213,10 +215,59 @@ func (c *controller) allocateIPs(key string, svc *v1.Service) ([]net.IP, error)
 		return ips, nil
 	}
 
+	pinnedPriorityPools, pinnedNoPriorityPools := c.getPinnedPools(svc)
```

Comment:
I'd move all this logic to allocator's `Allocate()`

Reply:
yes, this is a cleaner solution. moved the logic into `Allocate()`.